### ISSUE TITLE
Batching should also be added to descriptor computation

### ISSUE THREAD

**Issue by zd9999cs** [luigibonati/mlcolvar#182]
Currently, batching is added to `mlcolvar.core.loss.committor_loss.compute_descriptors_derivatives`(#181) , but the descriptor computation in it is not batched and can cause RAM overflow on a large dataset.

In `mlcolvar.core.loss.committor_loss.compute_descriptors_derivatives`

```
# we compute the descriptors on the whole dataset to always have all of them, no need for grads   
    with torch.no_grad():
        print("Computing descriptors on the whole dataset...")
        desc = descriptor_function(pos)                    # This can cause RAM overflow too
        print("Descriptors computed on the whole dataset!")
```

I suggest either of the following
- Add batching to `mlcolvar.core.transform.descriptors`
- Add batching to descriptor computation, and share the same batching passed to `mlcolvar.core.loss.committor_loss.compute_descriptors_derivatives`

**Comment by EnricoTrizio** [comment]
Ciao @zd9999cs can you quantify a "large dataset"?
